src/Sentry/Internal/AotHelper.cs

```diff
@@ -11,18 +11,14 @@ private class AotTester
 
     internal static bool IsAot { get; }
 
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = AotHelper.SuppressionJustification)]
     static AotHelper()
     {
-        try
-        {
-            // GetMethod should throw an exception if Trimming is enabled
-            var type = typeof(AotTester);
-            _ = type.GetMethod(nameof(AotTester.Test));
-        }
-        catch
-        {
-            IsAot = true;
-        }
+#if NET6_0_OR_GREATER   // TODO NET7 once we target it
+        var stackTrace = new StackTrace(false);
```

Comment:
This is kind of expensive I believe. Even though the Exception approach is definitely not cheap and kind of dirty. And fires `FirstChanceException` so definitely prefer not to use that approach